assets/js/project.js

```diff
@@ -0,0 +1,48 @@
+import start from '@asl/peephole';
+import debounce from 'lodash/debounce';
+
+const updateProject = project => {
+  return {
+    type: 'UPDATE_PROJECT',
+    project
+  };
+};
+
+const state = window.INITIAL_STATE;
+
+const postData = debounce(data => {
+  window.fetch(state.static.basename, {
+    method: 'PUT',
+    credentials: 'include',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify(data)
```

Comment:
We should use `r2` in place of native `fetch` to handle browsers that don't have `fetch`

Reply:
Also if you do that then you don't have to send the JSON headers, or `JSON.stringify` the data. You jus pass a `json` param.